### PR TITLE
Migrate to GHA

### PR DESCRIPTION
**Pull request by claire-labry**
This PR officially migrates `go-getter` to GHA from CircleCI. It's intention is to work the same as the CircleCI tests but in GHA format. 

Note: After working on updating the AWS/GCP tests, it was concluded that the older go versions (1.14.15 and 1.15.3) were blocking the GCP tests from passing. The issue was that go1.17 is required for the dependent GCS libraries and we concluded that the removal of the older go versions was the logical path to move forward.

### CODE REVIEW

**Comment by picatz** [comment]
Just to be clear, we will disable the CircleCI pipeline _after_ this PR is merged? 🤔

**Comment by claire-labry** [comment]
Yes, that's the intention! @picatz

File: get_s3_test.go
Comment (picatz) [inline get_s3_test.go:31]: We should probably remove the [`init`](https://github.com/hashicorp/go-getter/blob/d305097eb8901ba73041f9b4efcc8eae90f15dcd/get_s3_test.go#L12-L22) from this file, and document the expectation for how these tests should run (only in CI from PRs). This will be useful in the future, where external contributors will not be able to run these tests locally anymore, for example.
https://github.com/hashicorp/go-getter/blob/d305097eb8901ba73041f9b4efcc8eae90f15dcd/get_s3_test.go#L12-L22

File: get_gcs_test.go
Comment (picatz) [inline get_gcs_test.go:11]: Similar to my other comment for S3, we should documentation the expectation for where/how these tests should run. It won't be obvious if someone tries to run the tests locally for this module.

File: .github/workflows/go-getter.yml
Comment (picatz) [inline .github/workflows/go-getter.yml:58]: Optional formatting nit:

```suggestion
```

File: .github/workflows/go-getter.yml
Comment (picatz) [inline .github/workflows/go-getter.yml:127]: Optional formatting nit:

```suggestion
```

File: .github/workflows/go-getter.yml
Comment (picatz) [inline .github/workflows/go-getter.yml:35]: Optional consistency nit, different name from the other step name for Windows tests:

```suggestion
      - name: Setup cache for go modules
```

File: .github/workflows/go-getter.yml
Comment (picatz) [inline .github/workflows/go-getter.yml:15]: I suspect we'd be able to test `1.17`, `1.18`, and `1.19`, what do you think? 🤔 

```suggestion
        go-version:
          - 1.17
          - 1.18
          - 1.19
```

File: .github/workflows/go-getter.yml
Comment (picatz) [inline .github/workflows/go-getter.yml:94]: Similar to my suggestion in the Linux test steps, I suspect we'd be able to test `1.17`, `1.18`, and `1.19`?

```suggestion
        go-version:
          - 1.17
          - 1.18
          - 1.19
```

**Comment by claire-labry** [comment]
I'm going to defer to @dpowley for the first two comments on the s3 tests?

**Review by samsalisbury — COMMENTED** [review]
Thanks Claire, the caching seems to be broken atm, I left comments below... Other than that the CI bits all LGTM!

File: .github/workflows/go-getter.yml
Comment (claire-labry) [reply .github/workflows/go-getter.yml:45]: ah yeah that's my bad-- thank you for this tip!

**Comment by claire-labry** [comment]
@samsalisbury @picatz -- Can I get a re-review on this PR? The cache tests have been fixed -- @nywilken Would it be ok if we migrated [v2](https://github.com/hashicorp/go-getter/tree/v2) to GHA?

**Review by samsalisbury — APPROVED** [review]

**Comment by claire-labry** [comment]
@picatz would you be able to re-review this PR as I've added several things since your last review: [this commit here](https://github.com/hashicorp/go-getter/pull/379/commits/bad826d9c545eb209425c991226c487b0093641d) -- we need to allow the file to be transported globally due to the CVE linked in the [comments](https://github.com/hashicorp/go-getter/pull/379/commits/47ecbb0e0aae24d0304b71bea95a0f89153e0d2f). I also added comments to the GCS and S3 tests for external contributors on how they are able to run these tests. Once I have your final review and approval, I will merge this in.

File: get_s3_test.go
Comment (picatz) [inline get_s3_test.go:12]: Small formatting nit and typo fix.
```suggestion
// Note for external contributors: In order to run the S3 test suite, you will only be able to be run
```

File: get_gcs_test.go
Comment (picatz) [inline get_gcs_test.go:11]: Small formatting nit.
```suggestion
// Note for external contributors: In order to run the GCS test suite, you will only be able to be run
```

**Review by picatz — CHANGES_REQUESTED** [review]

File: get_s3_test.go
Comment (claire-labry) [reply get_s3_test.go:12]: 🤦🏻‍♀️ this is why i need more coffee! thank you

Comment (picatz) [reply get_s3_test.go]: 😄 ☕

**Review by picatz — APPROVED** [review]

**Pull request merged**
